go/x509util/x509util.go

```diff
@@ -0,0 +1,576 @@
+// Package x509util includes utility code for working with X.509
+// certificates from the x509 package.
+package x509util
+
```

Comment:
I hadn't done unit tests because it's purely for generating more friendly diagnostics/outputs -- it got tested by feeding several thousand real-world certificates through it.